inc/c_util/channel.h

```diff
@@ -40,11 +41,13 @@ typedef struct CHANNEL_TAG CHANNEL;
 
 THANDLE_TYPE_DECLARE(CHANNEL);
 
-    MOCKABLE_FUNCTION(, THANDLE(CHANNEL), channel_create, THANDLE(PTR(LOG_CONTEXT_HANDLE)), log_context, THANDLE(THREADPOOL), threadpool);
+    MOCKABLE_FUNCTION(, THANDLE(CHANNEL), channel_create, THANDLE(PTR(LOG_CONTEXT_HANDLE)), log_context, THANDLE(THREADPOOL), threadpool, uint32_t, channel_capacity);
```

Comment:
>channel_capacity [](http://example.com/codeflow?start=148&length=16)

Is this capacity in items? I assume so.
If it is in items it's not very useful by itself is it? Do we set it to some fixed number (like 100) and hope for the best?

A little birdie whispered in my ear that this is to control how much data we allow to push to the secondaries.

SF has 2 different parameters for controlling such queues:
item count and data size.

So, should we have also 2?

Reply:
Also this makes me have an uber comment whether the channel should ever do this shutting down of the pipe.
Channel is a decoupling concept as I understand it.

Now it gains a new responsibility. And this responsibility seems to grow in complexity if we also add multiple metrics to shut off the pipe on... It looks to me that we're a little bit violating the SRP. But just a little bit, so we pretend we don't :-D